zoobc/zoobc-core
# 1432 fix escrow and transaction fee calculations

## Description
Change the minimum fee calculation for transaction.
```
 fee = (txMessageMultiplier + escrowInstructionMultiplier + minFee) * escrowTimeout * feeScale
```

## Breakdown
- [x] Refactor the way transaction object is passed to the transactionType
- [x] Refactor fee calculation of transactions (to take transaction message, escrow instruction, and escrow timeout into account)
- [x] Remove usage of different fee instance (Normal Fee and Escrow Fee), instead using a util function to calculate the minimum transaction fee
- [x] Change initial fee scale to 0.01 ZBC

## Reference Issue
Close #1432

## Step to Test (optional)
- Run a node with variations of transaction message, escrow instruction, and escrow timeout
- Observe that the minimum fee required matches the requirement
